beesdoo_product/models/beesdoo_product.py

```diff
@@ -272,18 +299,28 @@ def _unit_same_category(self):
                     )
                 )
 
+    # fixme rename to _compute_suggested_price
+    # fixme move to new module product_suggested_price
+    #  or sale_suggested_price
     @api.multi
-    @api.depends("seller_ids", "supplier_taxes_id", "taxes_id")
+    @api.depends(
+        "seller_ids",
+        "supplier_taxes_id",
+        "taxes_id",
+        "uom_id",
+        "uom_po_id",
+        "categ_id.should_round_suggested_price",
+    )
     def _compute_cost(self):
         suggested_price_reference = (
             self.env["ir.config_parameter"]
             .sudo()
             .get_param("beesdoo_product.suggested_price_reference")
         )
         for product in self:
-            suppliers = product._get_main_supplier_info()
-            if len(suppliers) > 0:
-                price = suppliers[0].price
+            supplier = product._get_main_supplier_info()
+            if supplier:
```

Comment:
i would clearly favor an early exit here. 😉
```python
if not supplier:
    continue
```
of course, this would make the whole block appear in the diff.

Reply:
One point for you!